src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -68,7 +69,7 @@ public JsonTypeInfo BaseJsonTypeInfo
         public JsonNumberHandling? NumberHandling;
 
         // Required properties left
-        public HashSet<int>? RequiredPropertiesLeft;
+        public BitArray? RequiredPropertiesLeft;
```

Comment:
Nit: maybe it's just me, but calling it `RequiredPropertiesSet` and starting with a bitarray where everything is false (which is the default) sounds more intuitive.

Reply:
I would probably add a comment here clarifying what the boolean in each index signifies (i.e. "`ReadRequiredProperties[i]` indicates that `JsonTypeInfo.RequiredProperty[i]` has already been read by the current deserializer")

Reply:
honestly I'm in the other camp, I'll leave as is unless there is any perf or other reason to change it

Reply:
The `BitArray` class itself needs to do extra work to set all flags to `true`:

https://github.com/dotnet/runtime/blob/50d69de3eebb220565a80b46936a07ef6cbdc15a/src/libraries/System.Collections/src/System/Collections/BitArray.cs#L53-L63

It seems counterintuitive that the default value is all `true` rather than all `false`.